src/analysis/normed_space/bounded_linear_maps.lean

```diff
@@ -22,15 +23,162 @@ local notation f ` →_{`:50 a `} `:0 b := filter.tendsto f (nhds a) (nhds b)
 open filter (tendsto)
 open metric
 
+structure is_bounded_linear_map {k : Type*} [normed_field k]
+  {E : Type*} [normed_space k E] {F : Type*} [normed_space k F] (L : E → F)
+  extends is_linear_map k L : Prop :=
+(bound : ∃ M, M > 0 ∧ ∀ x : E, ∥L x∥ ≤ M * ∥x∥)
+
+-- refactor: bounded linear map as a bundled structure
+structure bounded_linear_map (k : Type*) [normed_field k]
+  (E : Type*) [normed_space k E] (F : Type*) [normed_space k F]
+  extends linear_map k E F :=
+(bounded : ∃ M, ∀ x : E, ∥to_fun x∥ ≤ M * ∥x∥)
+
 variables {k : Type*} [normed_field k]
 variables {E : Type*} [normed_space k E]
 variables {F : Type*} [normed_space k F]
 variables {G : Type*} [normed_space k G]
 
-structure is_bounded_linear_map {k : Type*}
-  [normed_field k] {E : Type*} [normed_space k E] {F : Type*} [normed_space k F] (L : E → F)
-  extends is_linear_map k L : Prop :=
-(bound : ∃ M, M > 0 ∧ ∀ x : E, ∥ L x ∥ ≤ M * ∥ x ∥)
+namespace bounded_linear_map
+
+instance : has_coe (bounded_linear_map k E F) (linear_map k E F) :=
+  ⟨to_linear_map⟩
+
+instance : has_coe_to_fun (bounded_linear_map k E F) :=
+  ⟨_, λ f, f.to_fun⟩
+
+@[extensionality] theorem ext {f g : bounded_linear_map k E F}
+  (h : ∀ x, f x = g x) : f = g :=
+  by cases f; cases g; congr' 1; ext x; apply h
+
+theorem ext_iff {f g : bounded_linear_map k E F} : f = g ↔ ∀ x, f x = g x :=
+  ⟨λ h x, by rintro; rw h, ext⟩
+
+variable {f : bounded_linear_map k E F}
+
+lemma map_zero : f 0 = 0 := linear_map.map_zero _
+
+lemma has_pos_bound : ∃ M > 0, ∀ x, ∥f x∥ ≤ M * ∥x∥ :=
+  let ⟨M, hf⟩ := f.bounded in
+  ⟨max 1 M, lt_of_lt_of_le zero_lt_one (le_max_left 1 _), λ _, le_trans
+  (hf _) (mul_le_mul_of_nonneg_right (le_max_right _ _) (norm_nonneg _))⟩
+
+-- some lemmas about things having bounds and things being bounds
+lemma ratio_has_pos_bound : ∃ M > 0, ∀ x, ∥f x∥ / ∥x∥ ≤ M :=
+  let ⟨M, hMp, hMb⟩ := has_pos_bound in ⟨M, hMp,
+  λ x, or.elim (lt_or_eq_of_le (norm_nonneg x))
+  (λ hlt, div_le_of_le_mul hlt (by rw mul_comm; exact hMb x))
+  (λ heq, by rw [←heq, div_zero]; exact le_of_lt hMp)⟩
+
+def of_linear_map_of_bounded {f : linear_map k E F}
+  (h : ∃ M, ∀ x, ∥f x∥ ≤ M * ∥x∥) : bounded_linear_map k E F := ⟨f, h⟩
+
+theorem is_bounded_linear_map {f: bounded_linear_map k E F} :
+  is_bounded_linear_map f :=
+  ⟨f.to_linear_map.is_linear,
+  let ⟨M, hMg, hMb⟩ := has_pos_bound in ⟨M, hMg, hMb⟩⟩
+
+-- the zero map and the identity maps are bounded
```

Comment:
If you want to write comments, that is very good. However, it would be even better to write docstrings. Because these docstrings will then show up in tooltips in VScode and such. The docstring has to be relevant for exactly the one statement that follows it. So this comment would have to be split into two docstrings.
You could also do this elsewhere in this PR.